saleor/search/views.py

```diff
@@ -8,21 +8,30 @@
 from ..product.utils import products_with_details, products_with_availability
 
 
-def paginate_results(results, get_data, paginate_by=25):
+def paginate_results(results, page_counter, total,
+                     paginate_by=settings.PAGINATE_BY):
     paginator = Paginator(results, paginate_by)
-    page_number = get_data.get('page', 1)
+    # TODO: Think about clean and nice solution to pagination problem
+    page_number = 1
     try:
         page = paginator.page(page_number)
+        page.number = page_counter
+        page.paginator.num_pages = total // settings.PAGINATE_BY + 1
```

Comment:
This is probably a bug when total number is a divisor of per-page number. You might use ex. .ceil().

Reply:
Good idea, thanks